`test_pathobject` fails with IsADirectoryError fsspec 0.8.3

When running all tests on our linux remote server the test `test_pathobject` fails with:

```bash
def test_pathobject():
        import pathlib
    
        here = os.path.abspath(os.path.dirname(__file__))
        flist = os.listdir(here)
        plist_str = [os.path.join(here, p) for p in flist]
        plist = [pathlib.Path(p) for p in plist_str]
        of = open_files(plist)
        assert len(of) == len(flist)
        assert [f.path for f in of] == plist_str
    
        of = open_files(plist[0])
        assert len(of) == 1
        assert of[0].path == plist_str[0]
>       with of[0] as f:

fsspec/tests/test_core.py:99: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
fsspec/core.py:102: in __enter__
    f = self.fs.open(self.path, mode=mode)
fsspec/spec.py:892: in open
    **kwargs
fsspec/implementations/local.py:115: in _open
    return LocalFileOpener(path, mode, fs=self, **kwargs)
fsspec/implementations/local.py:197: in __init__
    self._open()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <fsspec.implementations.local.LocalFileOpener object at 0x7ffff69a9b90>

    def _open(self):
        if self.f is None or self.f.closed:
            if self.autocommit or "w" not in self.mode:
>               self.f = open(self.path, mode=self.mode)
E               IsADirectoryError: [Errno 21] Is a directory: '/build/source/fsspec/tests/__pycache__'

fsspec/implementations/local.py:202: IsADirectoryError
```

If I run it locally on a OSX and linux popOS machine and on a remote OSX builder all tests pass. So I cannot reproduce it locally. This is strange as usually the linux builder passes all tests and the OSX fails😃.

Do you know what may cause this test to fail?

--- a/fsspec/tests/test_core.py
+++ b/fsspec/tests/test_core.py
@@ -82,15 +82,16 @@ def test_list():
     assert [f.path for f in of] == plist
 
 
-def test_pathobject():
+def test_pathobject(tmpdir):
     import pathlib
 
-    here = os.path.abspath(os.path.dirname(__file__))
-    flist = os.listdir(here)
-    plist_str = [os.path.join(here, p) for p in flist]
+    tmpdir = str(tmpdir)
+    plist_str = [os.path.join(str(tmpdir), f) for f in ["a", "b"]]
+    open(plist_str[0], "w").write('first file')
+    open(plist_str[1], "w").write('second file')
     plist = [pathlib.Path(p) for p in plist_str]
     of = open_files(plist)
-    assert len(of) == len(flist)
+    assert len(of) == 2
     assert [f.path for f in of] == plist_str
 
     of = open_files(plist[0])

--- a/fsspec/tests/test_core.py
+++ b/fsspec/tests/test_core.py
@@ -87,8 +87,8 @@ def test_pathobject(tmpdir):
 
     tmpdir = str(tmpdir)
     plist_str = [os.path.join(str(tmpdir), f) for f in ["a", "b"]]
-    open(plist_str[0], "w").write('first file')
-    open(plist_str[1], "w").write('second file')
+    open(plist_str[0], "w").write("first file")
+    open(plist_str[1], "w").write("second file")
     plist = [pathlib.Path(p) for p in plist_str]
     of = open_files(plist)
     assert len(of) == 2

